### PR TITLE
Use lazy formatting in logging calls

### PR DESCRIPTION
**Pull request by akorb**
See https://stackoverflow.com/q/34619790/2050020

### CODE REVIEW

File: scapy/packet.py [hunk 1]
```diff
@@ -988,9 +988,9 @@ def do_dissect_payload(self, s):
             except Exception:
                 if conf.debug_dissector:
                     if issubtype(cls, Packet):
-                        log_runtime.error("%s dissector failed" % cls.__name__)
+                        log_runtime.error("%s dissector failed", cls.__name__)
                     else:
-                        log_runtime.error("%s.guess_payload_class() returned [%s]" % (self.__class__.__name__, repr(cls)))  # noqa: E501
+                        log_runtime.error("%s.guess_payload_class() returned [%s]", self.__class__.__name__, repr(cls))  # noqa: E501
```
Comment (gpotter2) [inline scapy/packet.py:993]: Would be cool if you could get rid of the noqa while you're at it. Thanks

File: scapy/pipetool.py [hunk 2]
```diff
@@ -136,7 +136,7 @@ def run(self):
                         try:
                             fd.deliver()
                         except Exception as e:
-                            log_runtime.exception("piping from %s failed: %s" % (fd.name, e))  # noqa: E501
+                            log_runtime.exception("piping from %s failed: %s", fd.name, e)  # noqa: E501
```
Comment (gpotter2) [inline scapy/pipetool.py:139]: same

**Review by gpotter2 — APPROVED** [review]
Thanks for the PR. Sounds reasonable (though we're definetly not after this kind of performance 😄)

**Review by guedou — CHANGES_REQUESTED** [review]
Thanks for the PR! I have the comments as @gpotter2 regarding the noqa tags.

**Comment by akorb** [comment]
I applied the comments and changed some more code in the same fashion with `warning`

**Review by p-l- — APPROVED** [review]

**Comment by p-l-** [comment]
@guedou is it OK for you now?

**Pull request merged**